src/python/qeqiskit/backend/backend.py

```diff
@@ -89,6 +95,40 @@ def run_circuit_and_measure(self, circuit: Circuit, n_samples: int) -> Measureme
             raise ValueError("n_samples should be greater than 0.")
         return self.run_circuitset_and_measure([circuit], [n_samples])[0]
 
+    def run_circuitset_and_measure(
```

Comment:
[suggestion] For future reference, I think that it will be better if we kept "moving code around" refactor to separate PRs than "changing code's logic", as reviewing this is not a straightforward task right now :)

Reply:
Ok! It was just hard to write the code itself without doing some refactoring. 😅

Reply:
Oh absolutely! Guilty of that too many times to remember ;) 
But I think it would be a good practice.
Or perhaps – first make a PR with just reshuffling and then make another one with actual changes.
Or at least keep them in separate commits?

Reply:
good point. i definitely should have put them in separate commits at least. But i'll try to put them in separate PRs from here on.